src/EnergyPlus/PlantChillers.cc

```diff
@@ -938,13 +938,11 @@ namespace PlantChillers {
 
                 if (this->HeatRecSetPointNodeNum > 0) {
                     Real64 THeatRecSetPoint(0.0);
-                    {
-                        auto const SELECT_CASE_var(state.dataPlnt->PlantLoop(this->HRLoopNum).LoopDemandCalcScheme);
-                        if (SELECT_CASE_var == DataPlant::LoopDemandCalcScheme::SingleSetPoint) {
-                            THeatRecSetPoint = state.dataLoopNodes->Node(this->HeatRecSetPointNodeNum).TempSetPoint;
-                        } else if (SELECT_CASE_var == DataPlant::LoopDemandCalcScheme::DualSetPointDeadBand) {
-                            THeatRecSetPoint = state.dataLoopNodes->Node(this->HeatRecSetPointNodeNum).TempSetPointHi;
-                        }
+                    if (state.dataPlnt->PlantLoop(this->HRLoopNum).LoopDemandCalcScheme == DataPlant::LoopDemandCalcScheme::SingleSetPoint) {
+                        THeatRecSetPoint = state.dataLoopNodes->Node(this->HeatRecSetPointNodeNum).TempSetPoint;
+                    } else if (state.dataPlnt->PlantLoop(this->HRLoopNum).LoopDemandCalcScheme ==
+                               DataPlant::LoopDemandCalcScheme::DualSetPointDeadBand) {
+                        THeatRecSetPoint = state.dataLoopNodes->Node(this->HeatRecSetPointNodeNum).TempSetPointHi;
```

Comment:
These are fine, though I do wonder if there was discussion about making these into switch blocks?  The reason we have all these SELECT_CASE_var blocks is because in Fortran, the selectors could be integer, logical, or character, but in C++, the case labels have to be constant, non-extern, and cannot be strings.  In many cases within EnergyPlus, the case labels are either extern variables or strings, so it can't be made directly into a switch block, and you must use an IF structure.  But as we move toward more enumeration, the number of switch blocks we can use grows.

I am not proposing that you redo these as switch blocks now, but moving forward, I would personally like to see more switch blocks where they are possible.  I'm curious what others think though.  I didn't see a specific section title in the wiki that addressed this, but it could be in there if I poked around a little more.

In either case, this is definitely a huge improvement already, and thanks!

Reply:
The logic I used here was that: if there are only two comparisons, use if-else. If there are more than two, use Switch/case. Referring to this older discussion [here](https://github.com/NREL/EnergyPlus/pull/8706#discussion_r612616667). Will start moving to switch/case 👍🏽